### PR TITLE
fix #323: cyctl get nonexistent resource

### LINKED ISSUE #323: cyctl get nonexistent resource
**Describe the bug**
When fecthing modules using `cyctl`, you can specify which modules you want to get.
- `cyctl get modules` fetches all modules
- `cyctl get modules demo-1 demo-2` fetches only specified modules (demo-1 and demo-2)

But if you specify a module that doesn't exist, you get an empty response. When the specified module doesn't exist, `cyctl` should  print an error saying there is no module with the specified name - same as kubectl does:
```sh
kubectl get pods i-do-not-exist
Error from server (NotFound): pods "i-do-not-exist" not found
```

Code for fetching modules is [here](https://github.com/cyclops-ui/cyclops/blob/4aa1a59de4bd60aef422448ed6f740ff8509b218/cyctl/internal/get/modules.go#L44)

### PR DESCRIPTION
**Pull request by thebigbone**
closes #323 

## 📑 Description
Added a boolean check which will prevent the empty response if the specified module name doesn't exist.

## ✅ Checks
- [ ] I have updated the documentation as required
- [x] I have performed a self-review of my code

### CODE REVIEW

**Review by petar-cvit — CHANGES_REQUESTED** [review]
Hey @thebigbone, thanks for the PR!

I tried your solution and found that if you specify multiple modules that do not exist, it will print all module names in a single line:
```
./cyctl get modules not-exists another-not-exists
no module found with name: [not-exists another-not-exists]
NAME                 AGE
```
Can we print a line for each Module that does not exist?

Also, if none of the speficied Modules exist, we shouldn't print `NAME AGE` columns since they will be empty.

Another thing is that if you specify one Module that exists and another that doesn't, cyctl wont print that one Module does not exist.
```
./cyctl get modules not-exists exists
NAME                 AGE
exists               4m49s
```

**Comment by petar-cvit** [comment]
To cover all cases:
- specify all that do not exist
```
cyctl get modules not-exists another-not-exists
  
no module found with name: not-exists
no module found with name: another-not-exists
```

 - specify one that does exist and one that doesn't
  ```
  cyctl get modules not-exists exists

  no module found with name: not-exists
  NAME                 AGE
  exists                 28s
  ```

**Comment by thebigbone** [comment]
You're right. My approach was not covering all the possible cases. I have modified the implementation. Thanks for the feedback!

File: cyctl/internal/get/modules.go [hunk 1]
```diff
@@ -47,16 +47,34 @@ func listModules(clientset *client.CyclopsV1Alpha1Client, moduleNames []string)
 		for _, name := range moduleNames {
 			nameSet[name] = struct{}{}
 		}
-		filteredModules = []v1alpha1.Module{}
+		foundModules := []v1alpha1.Module{}
+		notFoundModules := []string{}
```
Comment (petar-cvit) [inline cyctl/internal/get/modules.go:51]: Could you initialize these with `make`?
```suggestion
		foundModules := make([]v1alpha1.Module, 0)
		notFoundModules := make([]string, 0)
```

**Review by petar-cvit — COMMENTED** [review]
Thanks @thebigbone! Remebered two more things while you are at it:
- Could you also exit with code 1 if there are any not found modules? You can call `os.Exit(1)`
- Also, could you add the same logic to `template_store.go` and `template_auth_rules.go` from the same folder?

**Comment by thebigbone** [comment]
> * Could you also exit with code 1 if there are any not found modules? You can call `os.Exit(1)`
Wouldn't his approach prevent it from showing the existing modules if there are multiple modules specified in the argument, and one of them might have the existing module

**Comment by thebigbone** [comment]
> @thebigbone I would still keep the same output (showing both existing and non existing modules), but would exit with 1 `if len(notFoundModules) > 0`. You would just need to extract `notFoundModules` from the scope of the if statement.
> 
> I see this useful when running cyctl in a script. You can call `cyctl get modules <name>` to see if a module exists by checking the exit status

Got it! I have added the same logic to `template_store.go` and `template_auth_rules.go`.

File: cyctl/internal/get/modules.go (hunk 1 above)
Comment (thebigbone) [reply cyctl/internal/get/modules.go]: Done. Also added the exit status.

File: cyctl/internal/get/template_store.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ var (
  cyctl get templates`
 )
 
+// listTemplate retrieves and displays a list of templatestore from the Cyclops API.
```
Comment (petar-cvit) [inline cyctl/internal/get/template_store.go:24]: can you remove the duplicated comment

File: cyctl/internal/get/template_auth_rules.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ var (
  cyctl get templateauthrule`
 )
 
+// listTemplateAuthRules retrieves and displays a list of templateauthrules from the Cyclops API.
```
Comment (petar-cvit) [inline cyctl/internal/get/template_auth_rules.go:24]: can you remove the duplicated comment

File: cyctl/internal/get/modules.go [hunk 4]
```diff
@@ -47,16 +48,36 @@ func listModules(clientset *client.CyclopsV1Alpha1Client, moduleNames []string)
...
+
 	}
 
 	headerSpacing := max(0, longestName-4)
-	fmt.Println("NAME" + strings.Repeat(" ", headerSpacing) + " AGE")
+	output := ""
+	if len(filteredModules) > 0 {
+		output += "NAME" + strings.Repeat(" ", headerSpacing) + " AGE\n"
+	} else {
+		os.Exit(1)
```
Comment (petar-cvit) [inline cyctl/internal/get/modules.go:77]: can you add this check to the end of the function so it prints out the existing modules (if any are found) and then exists with 1?

The output should still look like this:
```
cyctl get modules not-exists exists

no module found with name: not-exists
NAME                 AGE
exists                 28s
```
but should exit with 1

File: cyctl/internal/get/template_auth_rules.go
Comment (petar-cvit) [inline cyctl/internal/get/template_auth_rules.go:1]: Can you add exit status logic here as well?

File: cyctl/internal/get/template_store.go
Comment (petar-cvit) [inline cyctl/internal/get/template_store.go:1]: Can you add exit status logic here as well?

**Review by petar-cvit — CHANGES_REQUESTED** [review]

**Comment by thebigbone** [comment]
I have made the changes.

**Review by petar-cvit — APPROVED** [review]
@thebigbone thanks! 🧡

**Pull request merged**